content/en/blog/_posts/2024-08-19-pod-failure-policy-ga.md

```diff
@@ -0,0 +1,173 @@
+1.31, and how to use it in your Jobs.
+
+## About Pod Failure Policy
+
+When running a workload on Kubernetes, things can go wrong resulting in Pod
+failures. Still, you expect your workload to tolerate transient failures and
+continue towards successful completion.
+
+To compensate for transient issues, Kubernetes Jobs include the `backoffLimit`
+field, allowing you to specify the number of Pod failures tolerated during
+workload execution.
+
+However, relying solely on this field can lead to unnecessary operating costs,
+as setting a large `backoffLimit` value results in excessive Pod restarts.
+
+This becomes particularly problematic when running large-scale Jobs with
+thousands of long-running Pods across thousands of Nodes.
+
+The Pod Failure Policy extends the backoff limit mechanism to help you reduce
+costs in the following ways:
+
+- Gives you control to fail the Job as soon as a non-retriable Pod failure occurs.
+- Allows you to ignore retriable errors without increasing the `backoffLimit` field.
+
+For example, it allows you to run your workload on more affordable spot machines
+by ignoring Pod failures caused by
+[graceful node shutdown](/docs/concepts/cluster-administration/node-shutdown/#graceful-node-shutdown).
+
+The policy allows you to distinguish between retriable and non-retriable Pod
+failures based on container exit codes and Pod conditions of a failed Pod.
+
+## How it works
+
+You specify a Pod failure policy in the Job specification, represented as a list
+of rules.
+
+For each rule you define match requirements based on either:
+- Container exit codes using the `onExitCodes` property.
+- Pod conditions using the `onPodConditions` property.
+
+Additionally, for each rule, you specify one of the following actions to take
+when a Pod matches the rule:
+- `Ignore` - Do not count the failure towards the `backoffLimit` or `backoffLimitPerIndex`.
+- `FailJob` - Fail the entire Job and terminate all running Pods.
+- `FailIndex` - Fail the index corresponding to the failed Pod
+  [Backoff limit per index](/docs/concepts/workloads/controllers/job/#backoff-limit-per-index),
+- `Count` - Count the failure towards the `backoffLimit` or `backoffLimitPerIndex`.
+  This is the default behavior.
+
+When you run your Job and Pod failures occur, then Kubernetes matches the
+failed Pod status against the list of Pod failure policy rules, in the specified
+order, and takes the corresponding actions for the first matched rule.
+
+### Pod disruption conditions
+
+In order to allow matching Pod Failure Policy against failures due to
+disruptions initiated by Kubernetes, this feature comes with a new Pod
+condition, called `DisruptionTarget`.
+
+This condition is added by Kubernetes it case of retriable
+[disruption scenarios](/docs/concepts/workloads/pods/disruptions/#pod-disruption-conditions).
+The condition contains one of the following reasons corresponding to the
+scenarios:
```

Comment:
```suggestion
To allow matching Pod failure policy rules against failures caused by
disruptions initiated by Kubernetes, this feature introduces the `DisruptionTarget`
Pod condition.

This condition is added by Kubernetes when a Pod fails because of a retriable
[disruption scenario](/docs/concepts/workloads/pods/disruptions/#pod-disruption-conditions).
The `DisruptionTarget` condition contains one of the following reasons that corresponds to
these disruption scenarios:
```